### ISSUE TITLE
Strict version compatibility checks for plugins

### ISSUE THREAD

**Issue by Osiris-Team** [Osiris-Team/AutoPlug-Client#244]
**Importance:** 7/10

**Problem:**
Latest version of a plugin might not support currently running mc version.

**Solution:**
Check for version compatibility, like already done for mods.
This was never implemented for plugins since there weren't complaints.

**Comment by Zoriot** [comment]
@Osiris-Team what's the scope of it? For modrinth it is implemented

**Comment by Osiris-Team** [comment]
@Zoriot yeah, the idea is to also adapt this check to other APIs like spigot etc
